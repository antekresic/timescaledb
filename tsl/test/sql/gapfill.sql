-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.

-- Test gapfill with no data
CREATE TABLE i6385(time timestamptz NOT NULL, value float);
SELECT table_name FROM create_hypertable('i6385','time');

SELECT
  time_bucket_gapfill('1 day', time) AS time,
  sum(value) AS value
FROM
  i6385
WHERE
  time >= '2023-12-01 14:05:00+01'
  AND time < '2023-12-20 14:05:00+01'
GROUP BY 1
ORDER BY 1 DESC;

DROP TABLE i6385;
