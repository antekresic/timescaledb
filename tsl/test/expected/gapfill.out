-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.
-- Test gapfill with no data
CREATE TABLE i6385(time timestamptz NOT NULL, value float);
SELECT table_name FROM create_hypertable('i6385','time');
 table_name 
------------
 i6385
(1 row)

SELECT
  time_bucket_gapfill('1 day', time) AS time,
  sum(value) AS value
FROM
  i6385
WHERE
  time >= '2023-12-01 14:05:00+01'
  AND time < '2023-12-20 14:05:00+01'
GROUP BY 1
ORDER BY 1 DESC;
             time             | value 
------------------------------+-------
 Tue Dec 19 16:00:00 2023 PST |      
 Mon Dec 18 16:00:00 2023 PST |      
 Sun Dec 17 16:00:00 2023 PST |      
 Sat Dec 16 16:00:00 2023 PST |      
 Fri Dec 15 16:00:00 2023 PST |      
 Thu Dec 14 16:00:00 2023 PST |      
 Wed Dec 13 16:00:00 2023 PST |      
 Tue Dec 12 16:00:00 2023 PST |      
 Mon Dec 11 16:00:00 2023 PST |      
 Sun Dec 10 16:00:00 2023 PST |      
 Sat Dec 09 16:00:00 2023 PST |      
 Fri Dec 08 16:00:00 2023 PST |      
 Thu Dec 07 16:00:00 2023 PST |      
 Wed Dec 06 16:00:00 2023 PST |      
 Tue Dec 05 16:00:00 2023 PST |      
 Mon Dec 04 16:00:00 2023 PST |      
 Sun Dec 03 16:00:00 2023 PST |      
 Sat Dec 02 16:00:00 2023 PST |      
 Fri Dec 01 16:00:00 2023 PST |      
 Thu Nov 30 16:00:00 2023 PST |      
(20 rows)

DROP TABLE i6385;
